Sources/MessagingPushAPN/MessagingPushAPN+PushConfigs.swift

```diff
@@ -0,0 +1,35 @@
+import Foundation
+import UIKit
+
+extension MessagingPushAPN {
+    func setupAutoFetchDeviceToken() {
+        // Register for push notifications
+        UIApplication.shared.registerForRemoteNotifications()
+        // Swizzle method `didRegisterForRemoteNotificationsWithDeviceToken`
+        swizzleDidRegisterForRemoteNotifications()
+    }
+
+    private func swizzleDidRegisterForRemoteNotifications() {
+        let appDelegate = UIApplication.shared.delegate
+        let appDelegateClass: AnyClass? = object_getClass(appDelegate)
+        let originalSelector = #selector(UIApplicationDelegate.application(_:didRegisterForRemoteNotificationsWithDeviceToken:))
+        let swizzledSelector = #selector(application(_:didRegisterForRemoteNotificationsWithDeviceToken:))
+        swizzle(forOriginalClass: appDelegateClass, forSwizzledClass: MessagingPushAPN.self, original: originalSelector, new: swizzledSelector)
+    }
+
+    private func swizzle(forOriginalClass: AnyClass?, forSwizzledClass: AnyClass?, original: Selector, new: Selector) {
```

Comment:
[We have a swizzle function already in the SDK](https://github.com/customerio/customerio-ios/blob/b0f309f1da66a6717a48feba2a0edc7555c8b629/Sources/Tracking/CustomerIOImplementation%2BScreenViews.swift#L23-L26). Could you refactor the code so the code base only has 1 swizzle function that gets re-used?

Reply:
Yeah, I noted this too. I am wondering how can we achieve this with minimal efforts. One way is to put this code in Common module as `CioMethodSwizzle`, that can then be accessible through all other modules. But with this I think, MessagingPushAPN/MessagingPushFCM might still not be able to access CioMethodSwizzle class because there is no diGraph in APN/FCM module. 🤔 
Just thinking out loud, because now there 3 occurrences of swizzle.

Reply:
Good points. 

Because we are swizzling inside of a top-level class (`MessagingPushFCM`), directly referencing `DIGraph` instance from `SdkInitializedUtil` would be OK to do. 

Therefore, you could create a simple `Swizzle` class in the Common module, add that new class to the DiGraph, then inside of `MessagingPushFCM` classes, get a reference of `DiGraph` to then get an instance of this new Swizzle class. 

The `MessagingPush` module does have a [class that is a `ModuleTopLevelObject`](https://github.com/customerio/customerio-ios/blob/main/Sources/MessagingPush/MessagingPush.swift#L9) which has an instance of `DiGraph` ready to use. However, using that class in the `MessagingPushFCM` module might be complicated?

Reply:
As discussed, this can be taken care at a later point. Not a blocker for this PR.